limbo_console.gd

```diff
@@ -115,9 +125,44 @@ func _input(p_event: InputEvent) -> void:
 			get_viewport().set_input_as_handled()
 
 
+func _process(delta: float) -> void:
+	if _is_done_sliding: return
+
+	if _is_sliding_in:
+		_open_t = move_toward(_open_t, 1.0, _open_speed * delta)
+		if _open_t == 1:
+			_is_done_sliding = true
+	else: # We close faster than opening.
+		_open_t = move_toward(_open_t, 0.0, _open_speed * delta * 1.5)
+		if _open_t == 0:
+			_is_done_sliding = true
+
+	var eased := ease(_open_t, -1.75)
+	var new_y := remap(eased, 0, 1, -_control.size.y, 0)
+	_control.position.y = new_y
+
+	if _is_done_sliding and not _is_sliding_in:
+		_control.position.y = 0 # Reset in case of calling show_console() directly, so it isn't off screen.
+		hide_console()
+
+
 # *** PUBLIC INTERFACE
 
 
+func open_console() -> void:
+	if enabled:
+		_is_done_sliding = false
+		_is_sliding_in = true
+		show_console()
+
+
+func close_console() -> void:
+	if enabled:
+		_is_done_sliding = false
+		_is_sliding_in = false
+		# hide_console() is called in _process()
+
+
 func show_console() -> void:
```

Comment:
We shouldn't have both open/close and show/hide as it creates a confusing API. I suggest moving show/hide to private visibility, or inlining.